include/circt/Dialect/Handshake/HandshakeOps.td

```diff
@@ -226,16 +234,16 @@ def BufferOp : Handshake_Op<"buffer", [NoSideEffect, HasClock,
     TypeAttr:$dataType,
     Confined<I32Attr, [IntMinValue<1>]>:$size,
     AnyType:$operand,
-    BoolAttr:$sequential,
+    BufferTypeEnum:$bufferType,
     OptionalAttr<I64ArrayAttr>:$initValues);
   let results = (outs AnyType:$result);
   let skipDefaultBuilders = 1;
   let builders = [OpBuilder<(ins "Type":$dataType, "int":$size,
-    "Value":$operand, "bool":$sequential)>];
+    "Value":$operand, "StringRef":$bufferType)>];
 
   let extraClassDeclaration = [{
     bool isSequential() {
-      return (*this)->getAttrOfType<BoolAttr>("sequential").getValue();
+      return this->bufferType() == "seq";
```

Comment:
Looking at the generated code for a `StrEnumAttr`, it looks like it generates both string version of the attribute, as well as an underlying enum. To increase the safety of this code, i think it would be better to compare to these enum values instead of doing string-based comparison.
i.e., i'm expecting there to be generated enums called `BufferTypeEnum::seq` and `BufferTypeEnum::fifo` which you can pass around and compare to.